### PR TITLE
fix(go): invalid output for multi-line @return and @deprecated comments

### LINKED ISSUE #2457: Go: multi-line comments result in invalid go code
### Affected Languages

- [x] `golang`

### General Information
* **JSII Version:** 1.17.1

### What is the problem?

When generating go code from a module that includes multi-line comments (e.g. aws/constructs), the generated code is invalid.

For example, this typescript code:

```ts
  /**
   * All parent scopes of this construct.
   *
   * @returns a list of parent scopes. The last element in the list will always
   * be the current construct and the first element will be the root of the
   * tree.
   */
  public get scopes(): IConstruct[] { ... }
```

Results in this go code:

```go
	// All parent scopes of this construct.
	//
	// Returns: a list of parent scopes. The last element in the list will always
be the current construct and the first element will be the root of the
tree.
	Scopes []IConstructIface `json:"scopes"`
```

As you can see, we need to add `//` prefix to the multiline `@returns` docstring.

### Verbose Log

```
./constructs.go:457:8: syntax error: unexpected current, expecting semicolon or newline or }
./constructs.go:459:10: syntax error: unexpected ] after top level declaration
```

### PR DESCRIPTION
**Pull request by eladb**
Split multi-line comments and prefix with `//` so they are rendered correctly.

Fixes #2457

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii-calc/lib/documented.ts [hunk 1]
```diff
@@ -12,11 +12,11 @@ export class DocumentedClass {
   /**
    * Greet the indicated person.
    *
-   * This will print out a friendly greeting intended for
-   * the indicated person.
+   * This will print out a friendly greeting intended for the indicated person.
```
Comment (RomainMuller) [inline packages/jsii-calc/lib/documented.ts:15]: Is this intentional change?

File: packages/jsii-pacmak/lib/targets/go/documentation.ts [hunk 2]
```diff
@@ -49,13 +46,19 @@ export class Documentation {
...
       } else {
-        this.code.line(`// ${this.code.toPascalCase(stability)}.`);
+        this.emitComment(`${this.code.toPascalCase(stability)}.`);
       }
     }
   }
 
+  private emitComment(text = '') {
+    for (const line of text.split('\n')) {
+      this.code.line(`// ${line}`);
```
Comment (RomainMuller) [inline packages/jsii-pacmak/lib/targets/go/documentation.ts:58]:
```suggestion
      this.code.line(`// ${line}`.trim());
```

**Review by RomainMuller — COMMENTED** [review]
I am somehow irked by the trailing whitespace, which is so easy to get rid of...

File: packages/jsii-calc/lib/documented.ts (hunk 1 above)
Comment (eladb) [reply packages/jsii-calc/lib/documented.ts:15]: Yes, just wrapped to 100 chars.

File: packages/jsii-pacmak/lib/targets/go/documentation.ts (hunk 2 above)
Comment (eladb) [reply packages/jsii-pacmak/lib/targets/go/documentation.ts:58]: I've added this but then the snapshots actually changed much more dramatically so I decided to revert. I'll add.

**Review by RomainMuller — APPROVED** [review]

**Pull request merged**